### PR TITLE
docs: 730 docs add an index to the guide overview

### PR DESCRIPTION
**Pull request by davidberenstein1957**
- [X] add index page to how to guides
- [X] remove duplication from main index page

### CODE REVIEW

File: docs/sections/how_to_guides/index.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+    Steps are the building blocks of your pipeline. They can be used to generate data, evaluate models, or perform any other task.
+
+    [:octicons-arrow-right-24: Define Steps](basic/step/index.md)
+
+-   __Define Tasks that rely on LLMs__
+
+    ---
+
+    Tasks are a special type of step that rely on Language Models (LLMs) to generate data.
```
Comment (burtenshaw) [inline docs/sections/how_to_guides/index.md:21]:
```suggestion
    Tasks are a specific type of step that rely on Language Models (LLMs) to generate data.
```

File: docs/sections/how_to_guides/index.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+## Basic
+
+<div class="grid cards" markdown>
+
+-   __Define Steps for your Pipeline__
+
+    ---
+
+    Steps are the building blocks of your pipeline. They can be used to generate data, evaluate models, or perform any other task.
```
Comment (burtenshaw) [inline docs/sections/how_to_guides/index.md:13]:
```suggestion
    Steps are the building blocks of your pipeline. They can be used to generate data, evaluate models, manipulate data, or any other general task.
```

File: docs/sections/how_to_guides/index.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    Tasks are a special type of step that rely on Language Models (LLMs) to generate data.
+
+    [:octicons-arrow-right-24: Define Tasks](basic/task/index.md)
+
+-   __Define LLMs as local or remote models__
+
+    ---
+
+    LLMs are the core of your tasks. They can be used to outline behavior local or remote models.
+
```
Comment (burtenshaw) [inline docs/sections/how_to_guides/index.md:30]: It makes sense to remove APIs in the title, but I think that here's useful.

```suggestion
    LLMs are the core of your tasks. They are used to integrate with local models or remote APIs.

```

File: docs/sections/how_to_guides/index.md [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+    Distiset is a dataset object based on the datasets library that can be used to store and manipulate data.
+
+    [:octicons-arrow-right-24: Distiset](advanced/distiset.md)
+
+-  __Export data to Argilla__
+
+    ---
+
+    Argilla is a platform that can be used to store and share data.
+
```
Comment (burtenshaw) [inline docs/sections/how_to_guides/index.md:59]:
```suggestion
    Argilla is a platform that can be used to store, search, and apply feedback to datasets.
```

**Review by burtenshaw — COMMENTED** [review]
Just some small tweaks in the grid.

**Review by sdiazlor — APPROVED** [review]

**Pull request merged**